lib/src/sv_tlv.c

```diff
@@ -898,6 +898,7 @@ decode_signature(signed_video_t *self, const uint8_t *data, size_t data_size)
   uint8_t **signature_ptr = &verify_data->signature;
   uint8_t version = *data_ptr++;
   uint8_t encoding_status = *data_ptr++;
+  if(version < 2) (void)(*data_ptr++); //If the version is older than 2 move past the hash_type byte.
```

Comment:
I think it becomes more clear if you do this after the declarations, like
```
if (version < 2) {
  // Move past the written hash type since it is never used.
  data_ptr++;
}
```
or if you we should be really strict, you should check the hash_type value and make sure it equals 1.